lib/discovery.js

```diff
@@ -181,7 +188,14 @@ function mixinDiscovery(MySQL, mysql) {
         ' numeric_scale AS "dataScale",' +
         ' column_type AS "columnType",' +
         ' is_nullable = \'YES\' AS "nullable",' +
-        ' CASE WHEN extra LIKE \'%auto_increment%\' THEN 1 ELSE 0 END AS "generated"' +
+        `
+          case
```

Comment:
If there's `GENERATED ALWAYS` when using `VIRTUAL` or `STORED`, should we check for that? 

Also could you please make the syntax consistent? i.e. for line 193:
```
 WHEN extra LIKE '%virtual%' THEN 1
```